influxdb3_write/src/last_cache/mod.rs

```diff
@@ -204,14 +224,25 @@ impl LastCacheProvider {
             )
         };
 
-        // build a schema that only holds the field columns
-        let mut schema_builder = SchemaBuilder::new();
-        for (t, name) in schema
+        let mut schema_builder = ArrowSchemaBuilder::new();
+        // Add key columns first:
+        for (t, field) in schema
+            .iter()
+            .filter(|&(_, f)| key_columns.contains(f.name()))
+        {
+            if let InfluxColumnType::Tag = t {
+                // override tags with string type in the schema
```

Comment:
Add more context to this comment: it's because key values get represented as strings.